docs/user-guide/query/gremlin/gremlin.md

```diff
@@ -1,5 +1,83 @@
-# Gremlin Queries in Gaffer
+# Gremlin in Gaffer
+
+[Gremlin](https://tinkerpop.apache.org/gremlin.html) is a query language for traversing graphs.
+It is a core component of the Apache Tinkerpop library and allows users to easily express more complex graph queries.
+
+GafferPop is a lightweight Gaffer implementation of the [TinkerPop framework](https://tinkerpop.apache.org/), where TinkerPop methods are delegated to Gaffer graph operations.
 
 !!! warning
-    Gremlin support is currently limited and experimental in Gaffer. This page
-    is a work in progress to capture the current capabilities.
\ No newline at end of file
+    GafferPop is still in development and has some [limitations](gremlin-limits.md).
+    The implementation is basic and its performance is unknown in comparison to using standard Gaffer `OperationChains`.
+
+The addition of Gremlin as query language in Gaffer allows users to represent complex graph queries in a simpler language akin to other querying languages used in traditional and NoSQL databases.
+
+## Gremlin Features
+
+### Interfacing with Gremlin
+
+One of the great features of Gremlin is its versatility of use.
+There are a large number of supported language libraries that allow you to write queries in whichever coding language you prefer.
+For example, there is a [Python Gremlin](https://pypi.org/project/gremlinpython/) interface.
+This means your tooling won't have to change to write these queries which is a nice bonus for Gremlin.
+
+### Imperative & Declarative Queries
+
+Gremlin supports 3 main methods of querying methods that gives us an element of flexibility using the library.
+Imperative queries are procedural and describe what's happening at each sequential step whereas declarative queries describe what should happen but lets the query compiler decide how and which order steps should be ran in.
+Choosing the right method here allows for a lot of control over how our queries get ran or allows the controller to optimise.
+
+### OTLP and OLAP
+
+Gremlin queries are flexible in that they can be evaluated in a realtime (OLTP) or batch (OLAP) format, this allows us a lot of flexibility in use, especially when querying over a multi machine or federated graph.
+
+!!! tip
+    Information on Gremlin as a query language, its associated libraries and more in-depth tutorials can be found in the [Apache Tinkerpop Gremlin docs](https://tinkerpop.apache.org/gremlin.html).
+
+## Gremlin in Gaffer
+
+Gremlin was added to Gaffer as a new graph query language in version 2.1.
+There is a small demo on the [gaffer-docker repo](https://github.com/gchq/gaffer-docker/tree/develop/docker/gremlin-gaffer) using the "TinkerPop Modern" [demo graph](https://tinkerpop.apache.org/docs/current/images/tinkerpop-modern.png).
+
+## Basic Queries
```

Comment:
Might be worth altering this example slightly now that the recommended connection method is to a gremlin server, for example to connect to a gremlin server using the python api it would be like:

```python
from gremlin_python.process.anonymous_traversal_source import traversal

g = traversal().withRemote(
    DriverRemoteConnection('ws://localhost:8182/gremlin'))
```

Maybe a link to those docs too? e.g. https://tinkerpop.apache.org/docs/current/reference/#connecting-gremlin-server 

(Also I'm not sure these examples are in groovy I think they are the python api)